src/block/component/SelectionBoxComponent.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace customiesdevs\customies\block\component;
+
+use pocketmine\math\Vector3;
+use pocketmine\nbt\tag\CompoundTag;
+use pocketmine\nbt\tag\FloatTag;
+use pocketmine\nbt\tag\ListTag;
+
+class SelectionBoxComponent implements BlockComponent {
+
+	private bool $useSelectionBox;
+	private Vector3 $origin;
+	private Vector3 $size;
+
+	/**
+	 * Defines the area of the block that is selected by the player's cursor. If set to true, default values are used. If set to false, this block is not selectable by the player's cursor. If this component is omitted, default values are used.
+	 * @param Vector3 $origin MMinimal position of the bounds of the selection box. "origin" is specified as [x, y, z] and must be in the range (-8, 0, -8) to (8, 16, 8), inclusive.
+	 * @param Vector3 $size Size of each side of the selection box. Size is specified as [x, y, z]. "origin" + "size" must be in the range (-8, 0, -8) to (8, 16, 8), inclusive.
+	 * @param bool $useSelectionBox If Selection Should be Enabled, Default is set to `true`
```

Comment:
Would just move this line above the other `@param` docs to match the other, there's also a lot of capitals that are not needed
```diff
- * @param bool $useSelectionBox If Selection Should be Enabled, Default is set to `true`
+ * @param bool $useSelectionBox If selection should be enabled, default is set to `true`.
```